engine/src/main/java/com/odysseusinc/arachne/executionengine/service/impl/DescriptorServiceImpl.java

```diff
@@ -53,36 +60,64 @@ public DescriptorServiceImpl(RIsolatedRuntimeProperties rIsolatedRuntimeProps) {
         this(
                 new DescriptorBundle(rIsolatedRuntimeProps.getArchive(), new DefaultDescriptor()),
                 Optional.ofNullable(rIsolatedRuntimeProps.getArchiveFolder()).map(name -> new File(name).toPath()),
+                rIsolatedRuntimeProps.getDefaultDescriptorFile(),
                 rIsolatedRuntimeProps.isApplyRuntimeDependenciesComparisonLogic()
         );
     }
 
-    public DescriptorServiceImpl(DescriptorBundle defaultDescriptorBundle, Optional<Path> archiveFolder, boolean dependencyMatching) {
+    public DescriptorServiceImpl(DescriptorBundle defaultDescriptorBundle,
+                                 Optional<Path> archiveFolder,
+                                 boolean dependencyMatching
+    ) {
+        this(defaultDescriptorBundle, archiveFolder, null, dependencyMatching);
+    }
+
+    public DescriptorServiceImpl(
+            DescriptorBundle defaultDescriptorBundle,
+            Optional<Path> archiveFolder,
+            String defaultDescriptorFile,
+            boolean dependencyMatching
+    ) {
         this.defaultDescriptorBundle = defaultDescriptorBundle;
         this.archiveFolder = archiveFolder;
         this.dependencyMatching = dependencyMatching;
+        this.defaultDescriptorFile = defaultDescriptorFile;
     }
 
     @Override
     public Optional<List<Descriptor>> getDescriptors() {
+        if (defaultDescriptorFile == null) {
+            log.warn("*** Property runtimeservice.dist.defaultDescriptorFile is not set. Empty default descriptor will used.");
+        }
         return archiveFolder.map(dir -> {
             try (Stream<Path> files = Files.list(dir)) {
                 return files.map(Path::toFile).filter(file ->
-                        file.getName().startsWith(DESCRIPTOR_PREFIX) && file.isFile()
-                ).map(file -> {
-                    try (InputStream is = new FileInputStream(file)) {
-                        return mapper.readValue(is, Descriptor.class);
-                    } catch (IOException e) {
-                        throw new RuntimeException("Error getting descriptor from file: " + file.getName(), e);
-                    }
-                }).collect(Collectors.toList());
+                                file.getName().startsWith(DESCRIPTOR_PREFIX) && file.isFile()
+                        ).map(deserializeDescriptor())
+                        .peek(d -> {
+                            if (Objects.equals(d.getLeft().getName(), defaultDescriptorFile)) {
+                                this.defaultDescriptorBundle = toBundle(d.getRight());
```

Comment:
I'm confused with mutating defaultDescriptorBundle here. If it comes from a constructor, why does it change in a side-effect of a methods that looks like read-only operation?